docs/advanced_documentation/native-data-interface.md

```diff
@@ -62,13 +62,45 @@ node['id'] = [1, 2]
 node['u_rated'] = [150e3, 10e3]
 ```
 
+## Columnar data format
+
+We can also represent the contents mentioned `NodeInput` struct in [Structured Array](#structured-array) for only specific attributes.
+This is especially useful when the component in question is for example a transformer which has many default attributes in which case the user can save on a significant amount of memory. Hence we can term it into `NodeInputUrated` which is of `double` type. 
+(Note again, its representation in C++ is not similar to that of `NodeInputUrated`). 
+
+One can create a `std::vector<NodeInputUrated>` to hold input for multiple nodes.
+In a similar example we create attribute data with `u_rated` of two nodes of 150 kV and 10 kV.
+
+```c++
+std::vector<NodeInputUrated> node_u_rated_input{ 150e3 , 10e3 };
+```
+Similar would be the case for `NodeInputId` and `std::vector<NodeNodeInputId>`
+
+To recreate this in numpy array, we should create it with the specific dtype as  mentioned in [Structured Array](#structured-array) for each attribute.
+
+```python
+node_id = np.empty(shape=2, dtype=node_dtype["id"])
```

Comment:
```suggestion
node_dtype = power_grid_meta_data[DatasetType.input][ComponentType.node].dtype
node_id = np.empty(shape=2, dtype=node_dtype["id"])
```

Reply:
Maybe then explicitly mention it where the Meta-data Helper Module is brought up, together with the use of `initialize_array` for columnar data.